tests/test_views.py

```diff
@@ -980,3 +982,75 @@ def test_deletion_of_workspace_of_an_already_deleted_workflow(
                          hard_delete=False,
                          workspace=True)
         assert not os.path.exists(absolute_workflow_workspace)
+
+
+def test_get_workflow_diff(app, default_user,
+                           sample_yadage_workflow_in_db,
+                           sample_serial_workflow_in_db,
+                           tmp_shared_volume_path,
+                           sample_workflow_workspace):
+    """Test set workflow status for unknown workflow."""
+    with app.test_client() as client:
+        res = client.get(url_for(
+            'api.get_workflow_diff',
+            workflow_id_or_name_a=sample_serial_workflow_in_db.id_,
+            workflow_id_or_name_b=sample_yadage_workflow_in_db.id_),
+            query_string={'user': default_user.id_},
+            content_type='application/json')
+        assert res.status_code == 200
+        response_data = json.loads(res.get_data(as_text=True))
+        assert 'reana_specification' in response_data
+        assert 'workspace_listing' in response_data
+        workflow_diff = json.loads(response_data['reana_specification'])[
+            'workflow']
+        entire_diff_as_string = ''.join(str(e) for e in workflow_diff)
+        # the following should be present in the diff
+        assert 'serial' in ''.join(str(e) for e in json.loads(
+            response_data['reana_specification'])['workflow'])
+        assert 'yadage' in ''.join(str(e) for e in json.loads(
+            response_data['reana_specification'])['workflow'])
+        assert json.dumps(sample_serial_workflow_in_db.reana_specification[
+            'workflow']['specification']['steps'][0]['commands']) in \
+            entire_diff_as_string
+        # single line of the entire specification is tested
+        # get_workflow_diff() returns extra characters between lines
+        assert sample_yadage_workflow_in_db.reana_specification[
+            'workflow']['specification']['first'] in \
+            entire_diff_as_string
+        print('done')
+
+
+def test_get_workspace_diff(app, default_user,
+                            sample_yadage_workflow_in_db,
+                            sample_serial_workflow_in_db,
+                            tmp_shared_volume_path,
+                            sample_workflow_workspace):
+    """Test get workspace differences."""
+    # create the workspaces for the two workflows
+    workspace_path_a = next(sample_workflow_workspace(
+        str(sample_serial_workflow_in_db.id_)))
+    workspace_path_b = next(sample_workflow_workspace(
+        str(sample_yadage_workflow_in_db.id_)))
+
+    sample_serial_workflow_in_db.get_workspace = lambda: str(
```

Comment:
Why do we do this?

Reply:
This is interesting, in `rest.py` `workflow_instance.get_workspace()` will return a path like:
```
'/reana/users/user-id/workflows/workflow-id'
```
where in the tests the root directory is `/tmp/pytest-of-user/....` so it needs to be patched.

Reply:
mmm I am confused, [`get_workspace` always returns a relative path](https://github.com/reanahub/reana-db/blob/master/reana_db/utils.py#L13), then in the Flask app we [contextualize it to the `SHARED_VOLUME_PATH` value](https://github.com/reanahub/reana-workflow-controller/blob/master/reana_workflow_controller/rest.py#L377),  is it maybe the config variable that is not being correctly set?

Reply:
As you can see by running this test, `SHARED_VOLUME_PATH` is set correctly:
```
ipdb> workspace_a
'users/00000000-0000-0000-0000-000000000000/workflows/3df8c716-8fba-4fb5-ab7d-2d52500064e9'
ipdb> current_app.config['SHARED_VOLUME_PATH']
'/tmp/pytest-of-dinos/pytest-68/reana0'
```